src/main/java/runhistoryplus/patches/ImprovableCardRunHistoryPatch.java

```diff
@@ -0,0 +1,112 @@
+                if (improvableCards.contains(card.cardID)) {
+                    String metricID = card.getMetricID();
+                    if (!improvableCardsLog.containsKey(metricID)) {
+                        improvableCardsLog.put(metricID, new ArrayList<>());
+                    }
+                    improvableCardsLog.get(metricID).add(card.misc);
+                }
+            }
+            ReflectionHacks.privateMethod(Metrics.class, "addData", Object.class, Object.class)
+                    .invoke(__instance, "improvable_cards", improvableCardsLog);
+        }
+    }
+
+    @SpirePatch(clz = RunHistoryScreen.class, method = "update")
+    public static class RunHistoryScreenUpdatePatch {
+        @SpireInsertPatch(locator = Locator.class, localvars = "addMe")
+        public static void updateCardsWithFinalValues(RunHistoryScreen __instance, RunData ___viewedRun, TinyCard addMe) throws NoSuchFieldException, IllegalAccessException {
+            if (improvableCards.contains(addMe.card.cardID)) {
+                Field improvableCardsField = ___viewedRun.getClass().getField("improvable_cards");
+                Map<String, List<Double>> improvableCardsLog = (Map<String, List<Double>>)improvableCardsField.get(___viewedRun);
+                String metricID = addMe.card.getMetricID();
+                if (improvableCardsLog != null && improvableCardsLog.containsKey(metricID)) {
+                    Integer value = Collections.max(improvableCardsLog.get(metricID)).intValue();
+                    addMe.card.baseDamage = value;
+                    addMe.card.baseBlock = value;
+                    if (!addMe.card.rawDescription.contains(TEXT[0])) {
+                        ArrayList<Integer> values = new ArrayList<>();
+                        for (Double f : improvableCardsLog.get(metricID)) {
+                            values.add(f.intValue());
+                        }
+                        Collections.sort(values);
+                        addMe.card.rawDescription += TEXT[0] + values.toString();
+                        System.out.println(addMe.card.rawDescription);
+                        addMe.card.initializeDescription();
+                        for (int i = 0; i < addMe.card.description.size(); ++i) {
+                            System.out.println(addMe.card.description.get(i).text);
+                        }
+                    }
+                }
+            }
+        }
+
+        public static class Locator extends SpireInsertLocator {
+            public int[] Locate(CtBehavior ctMethodToPatch) throws CannotCompileException, PatchingException {
+                Matcher matcher = new Matcher.MethodCallMatcher(CardGroup.class, "addToTop");
+                return LineFinder.findInOrder(ctMethodToPatch, new ArrayList<>(), matcher);
+            }
+        }
+    }
+
+    @SpirePatch(clz = AbstractCard.class, method = "makeStatEquivalentCopy")
+    public static class AbstractCardCopyDescriptionPatch {
```

Comment:
I was expecting to see something like this given your comment about the card popup -- I'd forgotten how a lot of the card view screens make copies of cards. It's pretty annoying, really.

The core idea of what you're doing here makes sense, but the one worry I have is that we're patching `AbstractCard` itself, so this affects everything that uses `makeStatEquivalentCopy`. Even if we think our change is safe, there's a lot of places that use this method -- I see 51 from a quick search of the base game codebase. And there are potential issues with cards that change their description dynamically (like how Finisher shows the number of attacks played this turn, so you know how many times it will hit).

It's possible that this won't cause any problems -- but we'd have to do some thinking and testing to be sure, probably more than we really want to do.

Happily, there's a good alternative. Instead of patching `AbstractCard.makeStatEquivalentCopy`, we can patch `RunHistoryScreen.update`, and change what happens with the `CardCrawlGame.cardPopup.open` call. Specifically, right after that method call, we want to get the private `card` variable from `CardCrawlGame.cardPopup` and set its description to be the same as our card's description. It's more work, but it should be doable through an instrument patch followed by using `ReflectionHacks` to get the private variable.

Reply:
Unfortunately I wasn't able to get the instrument patch to work for now. A potential issue I see as well is, that this might only work for the card that you click on initially and not if you then use the left/right arrows to go through the deck while the card popup is still open.

As an alternative, I patched `SingleCardViewPopup.open` to copy the description there (since this is called by `openPrev()` and `openNext()`). This is again used in several places, but not nearly as many as `AbstractCard.makeStatEquivalentCopy`.

Reply:
Makes sense as another way to do it. When I have some time, I'll take another look over things, do some testing, and hopefully merge this in! Might take until next week.